one-offs/images-from-boms.go

```diff
@@ -0,0 +1,187 @@
+	Timestamp    string       `yaml:"timestamp"`
+	Services     Services     `yaml:"services"`
+	Dependencies Dependencies `yaml:"dependencies"`
+	//ArtifactSources ArtifactSources `yaml:"artifactSources"`
+}
+
+type Services struct {
+	Clouddriver      Service `yaml:"clouddriver"`
+	Deck             Service `yaml:"deck"`
+	Echo             Service `yaml:"echo"`
+	Fiat             Service `yaml:"fiat"`
+	Front50          Service `yaml:"front50"`
+	Gate             Service `yaml:"gate"`
+	Igor             Service `yaml:"igor"`
+	Kayenta          Service `yaml:"kayenta"`
+	MonitoringDaemon Service `yaml:"monitoring-daemon"`
+	Orca             Service `yaml:"orca"`
+	Rosco            Service `yaml:"rosco"`
+}
+
+func (s *Services) List() *list.List {
+	l := list.New()
+	l.PushBack(s.Clouddriver.WithName("clouddriver"))
+	l.PushBack(s.Deck.WithName("deck"))
+	l.PushBack(s.Echo.WithName("echo"))
+	l.PushBack(s.Fiat.WithName("fiat"))
+	l.PushBack(s.Front50.WithName("front50"))
+	l.PushBack(s.Gate.WithName("gate"))
+	l.PushBack(s.Igor.WithName("igor"))
+	l.PushBack(s.Kayenta.WithName("kayenta"))
+	l.PushBack(s.MonitoringDaemon.WithName("monitoring-daemon"))
+	l.PushBack(s.Orca.WithName("orca"))
+	l.PushBack(s.Rosco.WithName("rosco"))
+	return l
+}
+
+type Dependencies struct {
+	Consul Service `yaml:"consul"`
+	Redis  Service `yaml:"redis"`
+	Vault  Service `yaml:"vault"`
+}
+
+type Service struct {
+	Commit  string `yaml:"commit,omitempty"`
+	Version string `yaml:"version"`
+	name    string
+}
+
+func (s *Service) WithName(n string) *Service {
+	s.name = n
+	return s
+}
+
+//type ArtifactSources struct {
+//	DebianRepository   string `yaml:"debianRepository"`
+//	DockerRegistry     string `yaml:"dockerRegistry"`
+//	GitPrefix          string `yaml:"gitPrefix"`
+//	GoogleImageProject string `yaml:"gitPrefix,omitempty"`
+//}
+
+type Image struct {
+	Digest string   `json:"digest"`
+	Tags   []string `json:"tags"`
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+	storageSvc, err := storage.NewClient(ctx, option.WithCredentialsFile(*jsonKeyPath), option.WithScopes(storage.ScopeReadOnly))
+	if err != nil {
+		log.Fatalf("Error generating new storage client: %v", err)
+	}
+
+	boms := make([]*Bom, 0, 100)
+	iter := storageSvc.Bucket(*bucket).Objects(ctx, &storage.Query{Prefix: "bom/"})
+	for obj, err := iter.Next(); err == nil; obj, err = iter.Next() {
+		if releaseRegexp.Match([]byte(obj.Name)) {
+			log.Println(obj.Name)
+			r, err := storageSvc.Bucket(*bucket).Object(obj.Name).NewReader(ctx)
+			if err != nil {
+				log.Fatalf("error reading object %v: %v", obj.Name, err)
+			}
+			bom := &Bom{}
+			if err = yaml.NewDecoder(r).Decode(bom); err != nil {
+				log.Fatalf("error decoding bom %v: %v", obj.Name, err)
+			}
+			boms = append(boms, bom)
+			log.Printf("%+v", bom)
+			//break
+		}
+	}
+
+	// Go get all the images and index them by tag
+	imagesByTag := make(map[string]*Image, 1000)
+	l := boms[0].Services.List()
```

Comment:
If we're getting the services only from the first BOM, will this break depending on whether that one has all the services?  I'm thinking in particular of Kayenta, which isn't in the early BOMs, so depending on if we pick a BOM with kayenta here this might fail.

Reply:
The `List()` includes Kayenta - here's a snippet of the output:

```
2020/04/17 12:51:00 Executing gcloud for kayenta
2020/04/17 12:51:01 Indexed 197 images and 1053 tags for kayenta service
...
2020/04/17 12:51:06 Can't find kayenta: for bom 1.0.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.0.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.1.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.1.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.1.2
2020/04/17 12:51:06 Can't find kayenta: for bom 1.2.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.2.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.2.2
2020/04/17 12:51:06 Can't find kayenta: for bom 1.3.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.3.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.4.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.4.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.4.2
2020/04/17 12:51:06 Can't find kayenta: for bom 1.5.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.5.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.5.2
2020/04/17 12:51:06 Can't find kayenta: for bom 1.5.3
2020/04/17 12:51:06 Can't find kayenta: for bom 1.5.4
2020/04/17 12:51:06 Can't find kayenta: for bom 1.6.0
2020/04/17 12:51:06 Can't find kayenta: for bom 1.6.1
2020/04/17 12:51:06 Can't find kayenta: for bom 1.6.2
```
So it appears Kayenta was added either in 1.6.3 or 1.7.0

Reply:
Oh, I see, even if the first BOM doesn't have kayenta defined, the data type still does so it will be included in the iteration, 👍